onnxruntime/core/session/inference_session.cc

```diff
@@ -95,6 +95,8 @@ inline std::basic_string<T> GetCurrentTimeString() {
 
 }  // namespace
 
+std::atomic<uint32_t> InferenceSession::global_session_id_{1};
```

Comment:
Could 2 different users report the same session id in the events? How do we differentiate between them?

Reply:
yep.  by 2 users we could have:

- 2 apps on the same user desktop machine login instance
- 2 apps running on different machines

 the telemetry events have an "app session" global that is kept track of by the telemetry engines.    it lets us know which machine/app combo is reporting in.     so session_id is really the "ort session" differentiator inside a specific app instance (aka process ID).

---
In reply to: [335301237](https://github.com/microsoft/onnxruntime/pull/2109#discussion_r335301237) [](ancestors = 335301237)

Reply:
Is it possible to have 2 different sessions within the same app instance? If yes, can we still differentiate?

Reply:
In theory a long lived process could have global_session_id_ wrap around such that every 4B sessions would restart the numbering, and would maintain the same session id.    We actually have a state machine that lives in the cloud @ (xiangting) is helping us to lock down.  This is how we keep track of state and we can use to answer if the wrap around is ok, and also whether or not we need/want a "SessionEnd" event.  I'll file an open issue for this.

---
In reply to: [335779060](https://github.com/microsoft/onnxruntime/pull/2109#discussion_r335779060) [](ancestors = 335779060)

Reply:
Just to make sure I fully answers the quesiton :) is that what you meant?    Wrap around to 4 billion?  people can totally have 2 OrtSession's in the same app instance, and then we would differentiate using that uint32, right?  (assuming we are ok with wrap around).   another easy fix here is to just make it int64.

---
In reply to: [336570778](https://github.com/microsoft/onnxruntime/pull/2109#discussion_r336570778) [](ancestors = 336570778,335779060)